providers/core/core.battery.js

```diff
@@ -0,0 +1,78 @@
+/*globals navigator*/
+/*jshint node:true*/
+var PromiseCompat = require('es6-promise').Promise;
+var nav = navigator;
+
+var BatteryProvider = function(cap, dispatchEvent) {
+  "use strict";
+  this._batteryPromise = nav && nav.getBattery ? nav.getBattery() : null;
+
+  if (this._batteryPromise) {
+    this._batteryPromise.then(function(battery) {
+      this._setupListeners(battery, dispatchEvent);
+    }.bind(this));
+  }
+};
+
+BatteryProvider.prototype.getCharging = function() {
+  "use strict";
+  if (!this._batteryPromise) {
+    return PromiseCompat.resolve(true);  // Charging / plugged in
+  }
+  return this._batteryPromise.then(function(battery) {
```

Comment:
if i'm interpreting this correctly, the structure here means that creating a BatteryProvider object will run the promise for the getBattery() call. Is the resulting state object dynamic, or a snapshot of battery state at the time that the getBattery() promise is made?  (i guess, is it worth triggering `nav.getBattery()` to be called again in each of these methods?)